drivers/softlayer/driver_test.go

```diff
@@ -88,3 +90,9 @@ func TestSetConfigFromFlagsSetsImage(t *testing.T) {
 		t.Fatalf("expected 'MY_TEST_IMAGE'; received %s", img)
 	}
 }
+
+func TestHostnameDefaultsToMachineName(t *testing.T) {
+	d, _ := getTestDriver()
```

Comment:
Mostly to keep the test short, but I suppose it'll panic if there's an error since the driver will be nil. I'll handle the err instead.